### PR TITLE
Add fix to support jinja 3.1

### PR DESCRIPTION
**Pull request by melund**
This patches jinja2 support for 3.1. The cloud_sptheme library is currently abandoned so this patch is the only option currently. 

We should consider forking the library if the maintainer continues to be unresponsive. 

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by melund** [comment]
@conda-forge-admin, please rerender

File: recipe/0002-patch-jinja-markup-deprecation.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+diff --git a/cloud_sptheme/ext/index_styling.py b/cloud_sptheme/ext/index_styling.py
+index 8fd8e6d..68c9239 100644
+--- a/cloud_sptheme/ext/index_styling.py
++++ b/cloud_sptheme/ext/index_styling.py
+@@ -1,7 +1,7 @@
+ """cloud_sptheme.ext.index_styling - improves css styling for genindex"""
+ import logging; log = logging.getLogger(__name__)
+ import re
+-from jinja2 import Markup as literal, escape
++from markupsafe import Markup  as literal, escape
```
Comment (jakirkham) [inline recipe/0002-patch-jinja-markup-deprecation.patch]: nit: Seems to have an extra space here

```suggestion
+from markupsafe import Markup as literal, escape
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,10 +10,11 @@ source:
   sha256: caf47ac4c6346eef47fc11e799adbeaf645921c712cc099cf2530560e7ecde44
   patches:
     - 0001-patch-sphinx-4-support.patch
+    - 0002-patch-jinja-markup-deprecation.patch
```
Comment (jakirkham) [inline recipe/meta.yaml:13]: Have we tried PRing these [upstream]( https://foss.heptapod.net/doc-utils/cloud_sptheme )?

Comment (melund) [reply recipe/meta.yaml:13]: Yes. The library appears to be abandoned.  We haven't had any activity from the author the last year. 

https://foss.heptapod.net/doc-utils/cloud_sptheme/-/issues/45#note_158949
https://foss.heptapod.net/doc-utils/cloud_sptheme/-/issues/47

Maybe I should fork the repository on GitHub, but that would really be a new library.

**Comment by jakirkham** [comment]
Thanks Morten! 😄

**Pull request merged**